config/software/patch.rb

```diff
@@ -15,22 +15,25 @@
 #
 
 name "patch"
-default_version "2.7"
 
-version("2.7") { source md5: "1cbaa223ff4991be9fae8ec1d11fb5ab" }
-
-source url: "http://ftp.gnu.org/gnu/patch/patch-#{version}.tar.gz"
-
-relative_path "patch-#{version}"
+if windows?
+  default_version "2.6.1"
+  dependency "mingw-get"
+else
+  default_version "2.7"
+  version("2.7") { source md5: "1cbaa223ff4991be9fae8ec1d11fb5ab" }
+  source url: "http://ftp.gnu.org/gnu/patch/patch-#{version}.tar.gz"
+  relative_path "patch-#{version}"
+end
 
 env = with_standard_compiler_flags(with_embedded_path)
 
 build do
-  configure_command = ["./configure",
-                       "--prefix=#{install_dir}/embedded",
-                       "--disable-xattr"]
-
-  command configure_command.join(" "), env: env
-  make "-j #{workers}", env: env
-  make "-j #{workers} install", env: env
+  if windows?
+    command "mingw-get.exe -v install msys-patch-bin=#{version}-*", env:env
```

Comment:
It has a cli mode.  It's essentially the "bundler" of the mingw world.  `mingw-get install package>=major.minor.patch-platform-major.minor.patch" gets you a recent system library and any appropriate dependencies without having to list out every single package for "awk", "grep", "sed", "less" etc.  We already depend on the OS to provide sensible versions of those for the other platforms - so I figured it would be reasonable for us to do the same here.  If there's a security vulnerability, we have the ability to pin versions as needed.